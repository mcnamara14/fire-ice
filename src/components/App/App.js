import React, { Component } from 'react';
import PropTypes, { shape, func, string } from 'prop-types';
import logo from './logo.svg';
import './App.css';
import { connect } from 'react-redux';
import { fakeAction } from '../../actions';
import { fetchHouseData } from '../../apiCalls/index';
import { cleanHouseData } from '../../cleaners/cleanHouseData';
import { storeHouseData } from '../../actions/index';
import { Card } from '../Card/Card';

class App extends Component {
  constructor() {
    super();
    
    this.state = {
      loading: false
    }
  }

  async componentDidMount() {
    this.setState({
      loading: true
    });

    const data = await fetchHouseData();

    this.setState({
      loading: false
    });

    const cleanData = cleanHouseData(data);

    this.props.storeHouseData(cleanData);
  }

  render() {
    const houseCards = this.props.houseData.map((house, index) => {
      return (
        <Card key={index} {...house} />
      )
    })

    return (
      <div className='App'>
        <div className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <h2>Welcome to Westeros</h2>
        </div>
        <div className='Display-info'>
          { houseCards }
          { this.state.loading ? <div className="loading"></div> : null }
        </div>
      </div>
    );
  }
}

App.propTypes = {
  fake: shape({ fake: string }),
  fakeAction: func.isRequired
};

const mapStateToProps = state => ({
  houseData: state.houseData
});

const mapDispatchToProps = dispatch => ({ 
  storeHouseData: (data) => dispatch(storeHouseData(data))
});
export default connect(mapStateToProps, mapDispatchToProps)(App);
