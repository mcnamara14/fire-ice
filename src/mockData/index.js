export const mockCleanHouseData = [
  {
    ancestralWeapons: ["Lady Forlorn"],
    coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
    founded: "",
    name: "House Corbray of Heart's Home",
    seats: ["Heart's Home"],
    swornMembers: ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
    titles: ["Lord of Heart's Home", "King of the Fingers (historical)"],
    words: ""
  }
]

export const mockDirtyHouseData = [
  {
    ancestralWeapons: ["Lady Forlorn"],
    cadetBranches: [],
    coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
    founded: "",
    founder: "https://www.anapioficeandfire.com/api/characters/256",
    heir: "https://www.anapioficeandfire.com/api/characters/677",
    name: "House Corbray of Heart's Home",
    overlord: "https://www.anapioficeandfire.com/api/houses/7",
    region: "",
    seats: ["Heart's Home"],
    swornMembers: ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
    titles: ["Lord of Heart's Home", "King of the Fingers (historical)"],
    url: "https://www.anapioficeandfire.com/api/houses/81",
    words:""
  }
]