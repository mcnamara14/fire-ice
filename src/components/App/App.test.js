import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';
import * as mockData from '../../mockData/index';
jest.mock('../../cleaners/cleanHouseData');
jest.mock('../../apiCalls/index');
 
describe('App', () => {
  it('should match the snapshot', () => {
    const mockHouseData = mockData.mockHouseData;
    const wrapper = shallow(<App houseData={mockHouseData} storeHouseData={jest.fn()}/>);

    expect(wrapper).toMatchSnapshot();
  })
});