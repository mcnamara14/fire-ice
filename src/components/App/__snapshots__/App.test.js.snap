// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    houseData={
        Array [
            Object {
              "ancestralWeapons": Array [
                "Lady Forlorn",
              ],
              "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
              "founded": "",
              "name": "House Corbray of Heart's Home",
              "seats": Array [
                "Heart's Home",
              ],
              "swornMembers": Array [
                "https://www.anapioficeandfire.com/api/characters/255",
                "https://www.anapioficeandfire.com/api/characters/256",
                "https://www.anapioficeandfire.com/api/characters/447",
                "https://www.anapioficeandfire.com/api/characters/654",
                "https://www.anapioficeandfire.com/api/characters/677",
                "https://www.anapioficeandfire.com/api/characters/681",
                "https://www.anapioficeandfire.com/api/characters/871",
                "https://www.anapioficeandfire.com/api/characters/1182",
              ],
              "titles": Array [
                "Lord of Heart's Home",
                "King of the Fingers (historical)",
              ],
              "words": "",
            },
          ]
    }
    storeHouseData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="App-header"
>
          <img
                    alt="logo"
                    className="App-logo"
                    src="logo.svg"
          />
          <h2>
                    Welcome to Westeros
          </h2>
</div>,
        <div
          className="Display-info"
>
          <Connect(Card)
                    ancestralWeapons={
                              Array [
                                        "Lady Forlorn",
                                      ]
                    }
                    coatOfArms="Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)"
                    founded=""
                    name="House Corbray of Heart's Home"
                    seats={
                              Array [
                                        "Heart's Home",
                                      ]
                    }
                    swornMembers={
                              Array [
                                        "https://www.anapioficeandfire.com/api/characters/255",
                                        "https://www.anapioficeandfire.com/api/characters/256",
                                        "https://www.anapioficeandfire.com/api/characters/447",
                                        "https://www.anapioficeandfire.com/api/characters/654",
                                        "https://www.anapioficeandfire.com/api/characters/677",
                                        "https://www.anapioficeandfire.com/api/characters/681",
                                        "https://www.anapioficeandfire.com/api/characters/871",
                                        "https://www.anapioficeandfire.com/api/characters/1182",
                                      ]
                    }
                    titles={
                              Array [
                                        "Lord of Heart's Home",
                                        "King of the Fingers (historical)",
                                      ]
                    }
                    words=""
          />
          <div
                    className="loading"
          />
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
/>,
            <h2>
              Welcome to Westeros
</h2>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Westeros",
            },
            "ref": null,
            "rendered": "Welcome to Westeros",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <Connect(Card)
                ancestralWeapons={
                                Array [
                                                "Lady Forlorn",
                                              ]
                }
                coatOfArms="Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)"
                founded=""
                name="House Corbray of Heart's Home"
                seats={
                                Array [
                                                "Heart's Home",
                                              ]
                }
                swornMembers={
                                Array [
                                                "https://www.anapioficeandfire.com/api/characters/255",
                                                "https://www.anapioficeandfire.com/api/characters/256",
                                                "https://www.anapioficeandfire.com/api/characters/447",
                                                "https://www.anapioficeandfire.com/api/characters/654",
                                                "https://www.anapioficeandfire.com/api/characters/677",
                                                "https://www.anapioficeandfire.com/api/characters/681",
                                                "https://www.anapioficeandfire.com/api/characters/871",
                                                "https://www.anapioficeandfire.com/api/characters/1182",
                                              ]
                }
                titles={
                                Array [
                                                "Lord of Heart's Home",
                                                "King of the Fingers (historical)",
                                              ]
                }
                words=""
/>,
            ],
            <div
              className="loading"
/>,
          ],
          "className": "Display-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "ancestralWeapons": Array [
                "Lady Forlorn",
              ],
              "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
              "founded": "",
              "name": "House Corbray of Heart's Home",
              "seats": Array [
                "Heart's Home",
              ],
              "swornMembers": Array [
                "https://www.anapioficeandfire.com/api/characters/255",
                "https://www.anapioficeandfire.com/api/characters/256",
                "https://www.anapioficeandfire.com/api/characters/447",
                "https://www.anapioficeandfire.com/api/characters/654",
                "https://www.anapioficeandfire.com/api/characters/677",
                "https://www.anapioficeandfire.com/api/characters/681",
                "https://www.anapioficeandfire.com/api/characters/871",
                "https://www.anapioficeandfire.com/api/characters/1182",
              ],
              "titles": Array [
                "Lord of Heart's Home",
                "King of the Fingers (historical)",
              ],
              "words": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "loading",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="App-header"
>
            <img
                        alt="logo"
                        className="App-logo"
                        src="logo.svg"
            />
            <h2>
                        Welcome to Westeros
            </h2>
</div>,
          <div
            className="Display-info"
>
            <Connect(Card)
                        ancestralWeapons={
                                    Array [
                                                "Lady Forlorn",
                                              ]
                        }
                        coatOfArms="Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)"
                        founded=""
                        name="House Corbray of Heart's Home"
                        seats={
                                    Array [
                                                "Heart's Home",
                                              ]
                        }
                        swornMembers={
                                    Array [
                                                "https://www.anapioficeandfire.com/api/characters/255",
                                                "https://www.anapioficeandfire.com/api/characters/256",
                                                "https://www.anapioficeandfire.com/api/characters/447",
                                                "https://www.anapioficeandfire.com/api/characters/654",
                                                "https://www.anapioficeandfire.com/api/characters/677",
                                                "https://www.anapioficeandfire.com/api/characters/681",
                                                "https://www.anapioficeandfire.com/api/characters/871",
                                                "https://www.anapioficeandfire.com/api/characters/1182",
                                              ]
                        }
                        titles={
                                    Array [
                                                "Lord of Heart's Home",
                                                "King of the Fingers (historical)",
                                              ]
                        }
                        words=""
            />
            <div
                        className="loading"
            />
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
/>,
              <h2>
                Welcome to Westeros
</h2>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to Westeros",
              },
              "ref": null,
              "rendered": "Welcome to Westeros",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <Connect(Card)
                  ancestralWeapons={
                                    Array [
                                                      "Lady Forlorn",
                                                    ]
                  }
                  coatOfArms="Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)"
                  founded=""
                  name="House Corbray of Heart's Home"
                  seats={
                                    Array [
                                                      "Heart's Home",
                                                    ]
                  }
                  swornMembers={
                                    Array [
                                                      "https://www.anapioficeandfire.com/api/characters/255",
                                                      "https://www.anapioficeandfire.com/api/characters/256",
                                                      "https://www.anapioficeandfire.com/api/characters/447",
                                                      "https://www.anapioficeandfire.com/api/characters/654",
                                                      "https://www.anapioficeandfire.com/api/characters/677",
                                                      "https://www.anapioficeandfire.com/api/characters/681",
                                                      "https://www.anapioficeandfire.com/api/characters/871",
                                                      "https://www.anapioficeandfire.com/api/characters/1182",
                                                    ]
                  }
                  titles={
                                    Array [
                                                      "Lord of Heart's Home",
                                                      "King of the Fingers (historical)",
                                                    ]
                  }
                  words=""
/>,
              ],
              <div
                className="loading"
/>,
            ],
            "className": "Display-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "ancestralWeapons": Array [
                  "Lady Forlorn",
                ],
                "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
                "founded": "",
                "name": "House Corbray of Heart's Home",
                "seats": Array [
                  "Heart's Home",
                ],
                "swornMembers": Array [
                  "https://www.anapioficeandfire.com/api/characters/255",
                  "https://www.anapioficeandfire.com/api/characters/256",
                  "https://www.anapioficeandfire.com/api/characters/447",
                  "https://www.anapioficeandfire.com/api/characters/654",
                  "https://www.anapioficeandfire.com/api/characters/677",
                  "https://www.anapioficeandfire.com/api/characters/681",
                  "https://www.anapioficeandfire.com/api/characters/871",
                  "https://www.anapioficeandfire.com/api/characters/1182",
                ],
                "titles": Array [
                  "Lord of Heart's Home",
                  "King of the Fingers (historical)",
                ],
                "words": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "loading",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
